pkg/render/whisker/component.go

```diff
@@ -0,0 +1,318 @@
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/securitycontext"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+// The names of the components related to the Guardian related rendered objects.
+const (
+	WhiskerName               = "whisker"
+	WhiskerNamespace          = common.CalicoNamespace
+	WhiskerServiceAccountName = WhiskerName
+	WhiskerDeploymentName     = WhiskerName
+	WhiskerClusterRoleName    = WhiskerName
+
+	GoldmaneContainerName              = "goldmane"
+	WhiskerContainerName               = "whisker"
+	WhiskerBackendContainerName        = "whisker-backend"
+	ManagedClusterConnectionSecretName = "tigera-managed-cluster-connection"
+)
+
+func Whisker(cfg *Configuration) render.Component {
+	c := &Component{cfg: cfg}
+
+	return c
+}
+
+// Configuration contains all the config information needed to render the component.
+type Configuration struct {
+	PullSecrets           []*corev1.Secret
+	OpenShift             bool
+	Installation          *operatorv1.InstallationSpec
+	TrustedCertBundle     certificatemanagement.TrustedBundle
+	LinseedPublicCASecret *corev1.Secret
+}
+
+type Component struct {
+	cfg *Configuration
+
+	goldmaneImage       string
+	whiskerImage        string
+	whiskerBackendImage string
+}
+
+func (c *Component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.cfg.Installation.Registry
+	path := c.cfg.Installation.ImagePath
+	prefix := c.cfg.Installation.ImagePrefix
+
+	var err error
+
+	c.whiskerImage, err = components.GetReference(components.ComponentCalicoWhisker, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	c.whiskerBackendImage, err = components.GetReference(components.ComponentCalicoWhiskerBackend, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	c.goldmaneImage, err = components.GetReference(components.ComponentCalicoGoldmane, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *Component) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		render.CreateNamespace(WhiskerNamespace, c.cfg.Installation.KubernetesProvider, render.PSSRestricted, c.cfg.Installation.Azure),
+	}
+
+	objs = append(objs,
+		render.CreateOperatorSecretsRoleBinding(WhiskerNamespace),
+		c.serviceAccount(),
+		c.clusterRole(),
+		c.clusterRoleBinding(),
+		c.goldmaneService(),
+		c.whiskerService(),
+		c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
```

Comment:
This is not how we render the bundle elsewhere. I'd rather have us use the established pattern.